catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -237,6 +242,48 @@ public Fileset loadFileset(NameIdentifier ident) throws NoSuchFilesetException {
         });
   }
 
+  @Override
+  public FileInfo[] listFiles(NameIdentifier ident, String locationName, String subPath)
+      throws NoSuchFilesetException, IOException {
+    if (disableFSOps) {
+      LOG.warn("Filesystem operations disabled, rejecting listFiles for {}", ident);
+      throw new UnsupportedOperationException("Filesystem operations are disabled on this server");
+    }
+
+    String decodedSubPath = subPath;
+    if (StringUtils.isNotBlank(subPath)) {
+      decodedSubPath = URLDecoder.decode(subPath, StandardCharsets.UTF_8.name());
```

Comment:
could you plz move this decode operation to `filesetOperations.class` of the server module?